mimesis/providers/base.py

```diff
@@ -71,16 +71,16 @@ def __str__(self) -> str:
 class BaseDataProvider(BaseProvider):
     """This is a base class for all data providers."""
 
-    def __init__(self, locale: Optional[str] = None,
-                 seed: Optional[Seed] = None) -> None:
+    def __init__(self, locale: str = locales.DEFAULT_LOCALE,
+                 seed: Seed = None) -> None:
         """Initialize attributes for data providers.
 
         :param locale: Current locale.
         :param seed: Seed to all the random functions.
         """
         super().__init__(seed=seed)
         self._data: dict = {}
```

Comment:
Should be annotated as: `Dict[KeyType, ValueType`. `dict` means `Dict[Any, Any]`. And it does not help much.

Reply:
@sobolevn The problem is that we don't know `ValueType`.

Reply:
It may be `Any` in this case.

Reply:
It might be useful to enable `disallow_any_generics = True`